File: SparkDemo/src/main/scala/org/abondar/experimental/sparkdemo/scala/EmptyLineCount.scala
package org.abondar.experimental.sparkdemo.scala

import org.apache.spark.{SparkConf, SparkContext}


object EmptyLineCount {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setAppName("EmptyLineCount")
    val sc = new SparkContext(conf)

    val file = sc.textFile(args(0))
    val blankLines = sc.longAccumulator("lines")

    val callSigns = file.flatMap(line => {
      if (line == ""){
        blankLines.add(1)
      }
      line.split(" ")
    })

    callSigns.saveAsTextFile("/home/output")
    println("Blank lines: " + blankLines.value)
  }

}

File: SparkDemo/src/main/python/empty_line_count.py
from pyspark import SparkContext

import re, sys

sc = SparkContext("local", "Empty Lines")
file = sc.textFile(sys.argv[1])
blank_lines = sc.accumulator(0)

def extractCallSigns(line):
    global blank_lines
    if (line == ""):
        blank_lines +=1
    return line.split(" ")

call_signs = file.flatMap(extractCallSigns)
call_signs.saveAsTextFile(sys.argv[2] + "/callsigns")
print "Blank lines: %d" % blank_lines.value